CMS-Enterprise/easi-app
# [EASI-4630] Guidance letter frontend changes

# EASI-4630

## Description
- Update language to match [Figma](https://www.figma.com/design/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=6924-14527)
  - Language referring to the priority categories work will be updated in [EASI-4628](https://jiraent.cms.gov/browse/EASI-4628)
- Route updates:
  - Admin: /trb/:id/advice -> /trb/:id/guidance
  - Admin advice letter: /trb/:id/advice/:form-step -> /trb/:id/guidance/:form-step
  - Public advice letter: /trb/advice-letter/:id -> /trb/guidance-letter/:id
- Redirect to handle old routes
- Updated typenames, variable names, etc
- Updated unit tests and mock data

> [!NOTE]  
> Backend schema changes are still in progress. Frontend code still uses `adviceLetter` query names, enum types, etc. This will be updated before feature branch is merged into main.


## How to test this change
- Check advice letter admin section, advice letter form, and public advice letter to make sure language matches Figma
- Test routes redirect from

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
